frontend/tests/integration/components/transportation/tdf/modal-splits-test.js

```diff
@@ -225,4 +225,81 @@ module('Integration | Component | transportation/tdf/modal-splits', function(hoo
     // This action saves the transportationPlanningFactors object model, so check that server has correct value
     assert.equal(server.db.transportationPlanningFactors.firstObject.temporalModeSplits, true);
   });
+
+  test('user can toggle between Manual and Census Tracts tabs', async function(assert) {
+    const store = this.owner.lookup('service:store');
+
+    this.server.create('user');
+    this.server.create('data-package');
+    this.server.create('project', {
+      publicSchoolsAnalysis: this.server.create('publicSchoolsAnalysis'),
+      transportationAnalysis: this.server.create('transportationAnalysis', {
+        transportationPlanningFactors: [
+          this.server.create('transportationPlanningFactor', {
+            dataPackage: this.server.create('dataPackage', 'nycAcs'),
+            landUse: 'residential',
+            // necessary for displaying certain elements
+            // Temporal Splits tab (true) vs. All Periods tab (false)
+            temporalModeSplits: true,
+            // necessary for displaying certain elements
+            // user input mode splits (true) vs. mode split values from census tract calculator (false)
+            manualModeSplits: true,
+          }),
+        ],
+      }),
+    });
+
+    // define project model
+    const project = await store.findRecord('project', 1, {
+      include: 'transportation-analysis,transportation-analysis.transportation-planning-factors',
+    });
+
+    // replicating how availablePackages is defined on routes/project/show/transportation/tdf/planning-factors/show.js
+    const dataPackage = project.transportationAnalysis.get('transportationPlanningFactors').firstObject.get('dataPackage');
+    const availablePackages = await store.query('data-package', {
+      filter: {
+        package: dataPackage.get('package'),
+      },
+    });
+
+    this.project = project;
+    this.transportationPlanningFactorsResidentialModel = project.transportationAnalysis.get('transportationPlanningFactors').firstObject;
+    this.availablePackages = availablePackages;
+
+    await render(hbs`
+      {{#transportation/tdf/modal-splits
+        project=project
+        analysis=project.transportationAnalysis
+        factor=transportationPlanningFactorsResidentialModel
+        availablePackages=availablePackages}}
+      {{/transportation/tdf/modal-splits}}
+    `);
+
+    // #### CHECK DEFAULT VALUES #############################################
+    // check that correct elements are displayed when manualModeSplits = true
+    // total for am should be 3
+    assert.ok('[data-test-modal-split-input-am="auto"]');
+    // This action saves the transportationPlanningFactors object model, so check that server has correct value
+    assert.equal(server.db.transportationPlanningFactors.firstObject.manualModeSplits, true);
```

Comment:
Just an observation, open to discussion:

It's generally understood to be bad practice to assert against the mock backend state because it by necessity assumes knowledge about the structure of the data model itself (which would change). If we change the design of transportationPlanningFactors, these tests will no longer work, and the test itself would be less helpful because it wouldn't be asserting against the DOM state:

> In general we recommend asserting against Mirage's database and your UI, as the specifics of your app's HTTP requests should be considered implementation details of the behavior you're actually interested in verifying. But there's certainly valid reasons to drop down and assert against HTTP data.

(https://www.ember-cli-mirage.com/docs/testing/assertions#asserting-against-handled-requests-and-responses)

Reply:
Yes I agree!